app/src/main/java/com/codingblocks/cbonlineapp/viewmodels/VideoPlayerViewModel.kt

```diff
@@ -0,0 +1,82 @@
+package com.codingblocks.cbonlineapp.viewmodels
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.codingblocks.cbonlineapp.database.CourseDao
+import com.codingblocks.cbonlineapp.database.CourseRunDao
+import com.codingblocks.cbonlineapp.database.DoubtsDao
+import com.codingblocks.cbonlineapp.database.NotesDao
+import com.codingblocks.cbonlineapp.database.models.DoubtsModel
+import com.codingblocks.cbonlineapp.database.models.NotesModel
+import com.codingblocks.cbonlineapp.extensions.retrofitCallback
+import com.codingblocks.onlineapi.Clients
+import com.codingblocks.onlineapi.models.DoubtsJsonApi
+import com.codingblocks.onlineapi.models.Notes
+
+class VideoPlayerViewModel(
+    private var doubtsDao: DoubtsDao,
+    private var notesDao: NotesDao,
+    var courseDao: CourseDao,
+    var runDao: CourseRunDao
+) : ViewModel() {
+    var playWhenReady = false
+    var currentOrientation: Int = 0
+    var mOtp: String? = null
+    var mPlaybackInfo: String? = null
+
+    var getOtpProgress: MutableLiveData<Boolean> = MutableLiveData()
+    var createDoubtProgress: MutableLiveData<Boolean> = MutableLiveData()
+    var createNoteProgress: MutableLiveData<Boolean> = MutableLiveData()
+
+    fun getOtp(videoId: String, sectionId: String, attemptId: String) {
+        Clients.api.getOtp(videoId, sectionId, attemptId)
+            .enqueue(retrofitCallback { _, response ->
+                response?.let {
+                    if (it.isSuccessful) {
+                        mOtp = response.body()?.get("otp")?.asString
+                        mPlaybackInfo = response.body()?.get("playbackInfo")?.asString
+                    }
+                    getOtpProgress.value = (it.isSuccessful)
+                }
+            })
+    }
+
+    fun createDoubt(doubt: DoubtsJsonApi) {
+        Clients.onlineV2JsonApi.createDoubt(doubt)
+            .enqueue(retrofitCallback { _, response ->
+                response?.body()?.let {
+                    if (response.isSuccessful) {
+                        doubtsDao.insert(
+                            DoubtsModel(
+                                doubt.id, doubt.title, doubt.body, doubt.contents?.id
```

Comment:
it should be it.id as doubt.id is not auto-generated and is returned from the server.
Check the complete flow of the app make sure there is no crash/bug etc.

Reply:
Yes as I'm using let, I should use it here. Let me test the complete app again to verify everything is working. Give me an hour to make sure it is not glitched

Reply:
No, it's not about using let here doubt.id would be null as while creating an object of doubt we don't have an id it is generated from server side and then we save it do our DB

Reply:
same goes for notes as well

Reply:
also refractor VideoDoubtsFragment and VideoNotesFragment and delete the singleton object of appdatabase and make sure appdatabase object is being no where and only koin is being used for accessing database
Don't worry I'll accommodate points accordingly

Reply:
@aggrawalpulkit596, sounds great. This pr was very informative btw, thanks for having me here

Reply:
@aggarwalpulkit596 tell me a course which has doubts or notes in it

Reply:
Notes and doubts are user created thing not a course specific thing

Reply:
@aggarwalpulkit596 yes please name one course for me, I need to test them

Reply:
you can enroll in any course for trial and check the functionality

Reply:
@aggarwalpulkit596 great, let me try that